src/WorkBC.Importers.Federal/Services/XmlImportService.cs

```diff
@@ -22,33 +16,22 @@ public class XmlImportService
     {
         private const int MaxErrors = 25; // allow up to this many errors per run
         private readonly JobBoardContext _dbContext;
-        private readonly FederalSettings _federalSettings;
         private readonly Dictionary<ImportedJobFederal, DateTime> _lstUpdate =
             new Dictionary<ImportedJobFederal, DateTime>();
-        private readonly ProxySettings _proxySettings;
         private int _errorCount;
         private Dictionary<long, DateTime> _existingJobsDict = new Dictionary<long, DateTime>();
         private List<JobPosting> _lstInsert = new List<JobPosting>();
         private List<ImportedJobFederal> _lstPurge = new List<ImportedJobFederal>();
-        private XmlDocument _xmlDocumentEnglish = new XmlDocument();
-        private XmlDocument _xmlDocumentFrench = new XmlDocument();
         private readonly ILogger _logger;
         private readonly CommandLineOptions _options;
+        private readonly FederalApiService _apiService;
 
-        public XmlImportService(IConfiguration configuration, CommandLineOptions options, ILogger logger)
+        public XmlImportService(FederalApiService apiService, JobBoardContext dbContext, CommandLineOptions options, ILogger logger)
         {
-            string connectionString = configuration.GetConnectionString("DefaultConnection");
-            _dbContext = new JobBoardContext(connectionString);
+            _dbContext = dbContext;
             _logger = logger;
             _options = options;
-
-            //get API settings
-            _federalSettings = new FederalSettings();
-            configuration.GetSection("FederalSettings").Bind(_federalSettings);
-
-            //get Proxy settings
-            _proxySettings = new ProxySettings();
-            configuration.GetSection("ProxySettings").Bind(_proxySettings);
+            _apiService = apiService;
         }
```

Comment:
The constructor has been simplified and instead of passing in a configuration object, the required services are passed in which makes the service easier to test.